pkg/config/initoptions.go

```diff
@@ -142,19 +142,27 @@ func calcPackageNamespace(packageDir string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	// Return the first non-empty namespace found. Cluster-scoped
+	// Return the non-empty unique namespace if found. Cluster-scoped
 	// resources do not have namespace set.
+	currentNamespace := metav1.NamespaceNone
 	for _, node := range nodes {
 		rm, err := node.GetMeta()
-		if err != nil {
-			continue
+		if err != nil || len(rm.ObjectMeta.Namespace) == 0 {
```

Comment:
We shouldn't force customers to write the default namespace; we should just assume it.

Reply:
that's my original version, continue in this block and currentNamespace is initialized to metav1.NamespaceDefault.
in this case:
1. if namespace is not set, it defaults to "default"
2. only report error if there is a different namespace

while current setting:
1. always report error if no namespace is found in a resource, cluster resource included

Since the init does nothing than setting the namespace if inventory object, there seams no need to ask user to 
provide a namespace if none is found.
what you think @mortent ?

Reply:
updated, @seans3 please take a look